packages/worker/src/debug-replayer/client.ts

```diff
@@ -44,7 +44,9 @@ export class Client {
       request.once('response', resolve);
     });
     if (response.statusCode !== 200) {
-      throw new Error(`Bad response code from VS Code: ${response.statusCode}`);
+      const responseBody = await Client.readAll(response);
+      const message = JSON.parse(responseBody.toString())?.error ?? response.statusMessage;
```

Comment:
Oh I didn't realize we already had this `readAll` helper..

One last thing here.. Wrap the `JSON.parse` statement in `try .. catch`.
It's not safe to assume that all response bodies would contain valid JSON.

Reply:
Ah, good point. I think we can just ignore the error in that case? e.g.

```js
      let message = response.statusMessage;
      try {
        message = JSON.parse(responseBody.toString())?.error ?? message;
      } catch {
        // ignore
      }
```